sdks/go/pkg/beam/testing/passert/floats.go

```diff
@@ -26,6 +26,56 @@ import (
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
 )
 
+// EqualsFloat checks that two PCollections of floats are equal, with each element
+// being within a specified threshold of its corresponding element. Both PCollections
+// are loaded into memory, sorted, and compared element by element.
+func EqualsFloat(s beam.Scope, observed, expected beam.PCollection, threshold float64) {
+	s = s.Scope(fmt.Sprintf("passert.EqualsFloat[%v]", threshold))
+	beam.ParDo0(s, &thresholdFn{threshold: threshold}, beam.Impulse(s), beam.SideInput{Input: observed}, beam.SideInput{Input: expected})
+}
+
+type thresholdFn struct {
+	threshold float64
+}
+
+func (f *thresholdFn) ProcessElement(_ []byte, observed, expected func(*beam.T) bool) error {
+	var observedValues, expectedValues []float64
+	var observedInput, expectedInput beam.T
+	for observed(&observedInput) {
+		val := reflect.ValueOf(observedInput.(interface{})).Convert(reflectx.Float64).Interface().(float64)
```

Comment:
Since we now have 4 places where this reflective snippet is used, consider writing a "toFloat" function to abstract it, making the calling code easier to read.

Reply:
Done